### ISSUE TITLE
Generative zarr example / reusing zarr-related plugins

### ISSUE THREAD

**Issue by manzt** [napari/napari#1524]
## 🚀 Feature
Add mandelbrot set (or other generative multiscale) example to docs to showcase napari's multiscale capabilities.

## Motivation
Current multiscale examples are either toy examples where the whole pyramid fits in memory (tiled astronaut) or require download / access to some large remote dataset. This usually means quality internet access is necessary when trying to showcase the extent of napari's multiscale handling to newcomers. 

## Pitch

Provide a simple, fun, and contained multiscale generative example that can be run locally. Since zarr is designed to be hackable, this can be done with a special store. I implemented the mandelbrot set as a multiscale zarr using `numba` to generate chunks quickly -- [gist](https://gist.github.com/manzt/9b67d61e7adae3416bef91669bfd0acf).

![mandelbrot](https://user-images.githubusercontent.com/24403730/89442109-3d8e1800-d71c-11ea-9c82-0444a9e370bb.gif)

## Alternatives
A much more simple example is to just keep generating the same tiles at each zoom level for a generic image -- [gist](https://gist.github.com/aa7adc3b8c797133653a018fc9d5fff3), but not nearly as fun :)

## Additional context

Following up from sync today -- @tlambert03 @sofroniewn 

One final thought around `zarr` + lazy loading in napari is that it would be great to reuse the different zarr store implementations for to enable flexible ways of accessing a store by napari and other tools. The zarr-based reader plugins I've implemented for napari follow this pattern in the `reader_function`:

- Initialize a special store (`MutableMapping`) that wraps a local multiscale image format or generative function
- Create a list of `dask.array`'s using the paths included the multiscales extensions
- Return layer list

This works great for loading the data locally into napari from various file formats, but plugins of this type don't work for remote data that needs to be accessed over HTTP. What I want to propose is some way to reuse the `get_napari_reader` from these plugins and some standard (optional) way of returning an initialized zarr store.

### Concrete example
Remote server has large pyramidal tiff that can be read with `napari-lazy-openslide`. Howewer, I want to view this image locally in napari, how do I do that? The current option for multiscale non-zarr (if I understand correctly) is to download the data locally and use something like the `napari-lazy-openslide` plugin.

However, I can write a custom script using something like [`simple-zarr-server`](https://github.com/manzt/simple-zarr-server) that initializes a custom store for the TIFF on the server, and then exposes the zarr over HTTP (providing remote access). Since the open slide store implements the multiscales specification, it can be read remotely using other napari plugins (`ome-zarr-py`). 

#### Server
```python
import zarr
from simple_zarr_server import serve
from napari_lazy_openslide import OpenSlideStore

if __name__ == "__main__":
  store = OpenSlideStore('data/tumor_04.tif') # intitialize pyramidal tiff store on server
  grp = zarr.open(store)
  serve(grp) # exposes the zarr store over http
```

#### Client

```bash
$ napari http://my_remote_url.io/data.zarr # Uses something like ome-zarr-py plugin to read
```

It would be great to have the `napari_get_reader` function and some type of zarr store initialization (for readers that implement this pattern) which can be used by something like `simple-zarr-server`, enabling:

```bash
$ simple-zarr-server ./tumor_04.tif # uses napari reader plugin to see if it can initialize a store.
```

This could look like a special function for plugins of this type, e.g. `initialize_zarr_store(path)`, that gets called within `reader_function` for the plugin, but could by imported by other python libraries that just want the zarr object (not dask or layers list). In this case, the third-party library could check the `napari_get_reader` return to see if the plugin returns a reader, and then try to import the `initialize_zarr_store` for the plugin. @jni Standardizing this type of zarr store creation (for certain plugins) could also be useful if napari ends up coming up with an official way of serving data remotely. I think we discussed a magic jupyter command on zulip:
```
%serve ./tumor_04.tiff
```

**Comment by jni** [comment]
@manzt I'm not sure why you've conflated the Mandelbrot set and the generic remote zarr store in the same issue, but both are *super cool* and please add the Mandelbrot example to our examples NOW. =D

**Comment by manzt** [comment]
Yeah ... originally the generic zarr stuff was a minor comment / observation about zarr-based images that came up in the sync today and I wanted to put somewhere. Turns out I had a lot more to say apparently ... sorry 😅 Will move to a separate issue.

**Comment by kephale** [comment]
Oh my goodness, @manzt I just found this issue. I'm going to follow up on our other generative zarr discussion, but I'm linking this to the new Mandelbulb demo (which I am still cleaning up, but grew from your Mandelbrot demo):

https://github.com/napari/napari/issues/5561#issuecomment-1598974885

**Comment by manzt** [comment]
oh wow, I completely forgot about this PR! That's a little embarrassing. Time flies...
